lib/route-recognizer.js

```diff
@@ -164,6 +164,9 @@ function State(charSpec) {
   this.charSpec = charSpec;
   this.nextStates = [];
```

Comment:
it may be worth exploring making these nextStates and charSpecs lazy allocated, but that would depend on what % of stats they are used for.